frontend/src/pages/PipelineDetails.tsx

```diff
@@ -372,47 +445,99 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
     } else {
       // if fromRunId is not specified, then we have a full pipeline
       const pipelineId = this.props.match.params[RouteParams.pipelineId];
-      let templateResponse: ApiGetTemplateResponse;
 
       try {
         pipeline = await Apis.pipelineServiceApi.getPipeline(pipelineId);
-        pageTitle = pipeline.name!;
       } catch (err) {
         await this.showPageError('Cannot retrieve pipeline details.', err);
         logger.error('Cannot retrieve pipeline details.', err);
         return;
       }
 
+      const versionId = this.props.match.params[RouteParams.pipelineVersionId];
+
       try {
-        templateResponse = await Apis.pipelineServiceApi.getTemplate(pipelineId);
-        templateString = templateResponse.template || '';
+        // TODO(rjbauer): it's possible we might not have a version, even default
+        if (versionId) {
+          version = await Apis.pipelineServiceApi.getPipelineVersion(versionId);
+        }
       } catch (err) {
-        await this.showPageError('Cannot retrieve pipeline template.', err);
-        logger.error('Cannot retrieve pipeline details.', err);
+        await this.showPageError('Cannot retrieve pipeline version.', err);
+        logger.error('Cannot retrieve pipeline version.', err);
         return;
       }
 
+      selectedVersion = versionId ? version! : pipeline.default_version;
+
+      if (!selectedVersion) {
+        // An empty pipeline, which doesn't have any version.
+        pageTitle = pipeline.name!;
+        const actions = this.props.toolbarProps.actions;
+        actions[ButtonKeys.DELETE_RUN].disabled = true;
+        this.props.updateToolbar({ actions });
+      } else {
+        // Fetch manifest for the selected version under this pipeline.
+        pageTitle = pipeline.name!.concat(' (', selectedVersion!.name!, ')');
+        try {
+          // TODO(jingzhang36): pagination not proper here. so if many versions,
+          // the page size value should be?
+          versions =
+            (await Apis.pipelineServiceApi.listPipelineVersions('PIPELINE', pipelineId, 50))
```

Comment:
Just double check, does this return newest 50 versions? I think behavior is acceptable in this case.

Reply:
I believe the backend default is asc, if nothing specified. So oldest first. You prefer desc? I changed it.

Reply:
Newest sounds good. I believe usually people check latest and near-latest versions